File: backend/dataset/coinChange/coinChange_11.py
class Solution:
    def coinChange(self, coins: List[int], amt: int) -> int:
        dp = [[0 for i in range(amt+1)]for i in range(len(coins)+1)]
        for j in range(amt+1):
            dp[0][j] = sys.maxsize
        for i in range(1,len(coins)+1):
            for j in range(1,amt+1):
                if coins[i-1] > j:
                    dp[i][j] = dp[i-1][j]
                else:
                    dp[i][j] = min(1+dp[i][j-coins[i-1]],dp[i-1][j])
        if dp[-1][-1] != sys.maxsize:
            return dp[-1][-1]
        return -1

File: backend/CDataset/coinChange/coinChange_11.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int> dp(amount+1,amount+1);
        dp[0] = 0;
        for(auto x:coins)
            for(int i=x; i<=amount; i++)
                dp[i] = min(dp[i],dp[i-x]+1);
        int ans = -1;
        if(dp[amount]!=amount+1) ans = dp[amount];
        return ans;
    }
};
